src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.ImportWorkloads.targets

```diff
@@ -35,4 +35,27 @@ Copyright (c) .NET Foundation. All rights reserved.
                           NETCoreSdkVersion="$(NETCoreSdkVersion)"
                           GenerateErrorsForMissingWorkloads="true"/>
   </Target>
+
+  <!--  This target is not part of the build. Only used by dotnet workload restore command. Global property "SkipResolvePackageAssets"
+        need to be set to "true" to avoid requiring restore.-->
+  <Target Name="_FindAllReferenceWorkload" DependsOnTargets="GetSuggestedWorkloads;PrepareProjectReferences" Returns="@(_ResolvedSuggestedWorkloads)">
+    <MSBuild
+      Projects="@(_MSBuildProjectReferenceExistent)"
+      Targets="GetSuggestedWorkloads"
+      BuildInParallel="$(BuildInParallel)"
+      Properties="GenerateErrorsForMissingWorkloads=false; SkipResolvePackageAssets=true"
+      Condition="'%(_MSBuildProjectReferenceExistent.BuildReference)' == 'true' and '@(ProjectReferenceWithConfiguration)' != '' and '@(_MSBuildProjectReferenceExistent)' != ''"
+      RemoveProperties="%(_MSBuildProjectReferenceExistent.GlobalPropertiesToRemove)$(_GlobalPropertiesToRemoveFromProjectReferences)">
```

Comment:
I think we also need to remove `TargetFramework` here.  Otherwise, I think we'll end up doing the wrong thing in the following scenario:

Project A multitargets `net6.0-ios;net6.0-android`, and references Project B.
Project B targets `net6.0`.

If we don't remove the `TargetFramework`, then I think we'd end up passing `net6.0-ios` and `net6.0-android` as the `TargetFramework` to project B when the project A inner builds traverse the project references.  So we'd end up building the project with a TargetFramework that it doesn't target.

Reply:
Good catch. Removed